### PR TITLE
fix: ci deprecated warnings

### LINKED ISSUE #4444: Fix CI deprecated warnings
**What would you like to be added/modified**:
There're some deprecated warnings, need fixing
https://github.com/kubeedge/kubeedge/actions/runs/3545349196
such as
```
Unit test, integration test edgeNode.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/setup-go@v2, actions/cache@v2, actions/checkout@v2Show less
--
Unit test, integration test edgeThe `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```
/help
/good-first-issue
/good first issue
**Why is this needed**:

### PR DESCRIPTION
**Pull request by YTGhost**
Signed-off-by: HIHIA <283304489@qq.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Fix CI deprecated warnings
**Which issue(s) this PR fixes**:

Fixes #4444 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: .github/workflows/build-tools.yml [hunk 1]
```diff
@@ -10,10 +10,10 @@ on:
 jobs:
   publish-image-to-dockerhub:
     name: publish to DockerHub
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-22.04
```
Comment (gy95) [inline .github/workflows/build-tools.yml:13]: Thanks for your contribution :)
do we also need to update ubuntu version? or we can just still use 18.04.
@YTGhost

Comment (YTGhost) [reply .github/workflows/build-tools.yml:13]: @gy95 The ubuntu-18.04 environment is also deprecated, we should switch to ubuntu-20.04 or ubuntu-22.04 instead. see here https://github.com/actions/virtual-environments/issues/6002

Comment (gy95) [reply .github/workflows/build-tools.yml:13]: ok, waiting for maintainers to approve running CI :)

Comment (YTGhost) [reply .github/workflows/build-tools.yml:13]: @gy95 PTAL

Comment (gy95) [reply .github/workflows/build-tools.yml:13]: there're still two left warnings: https://github.com/kubeedge/kubeedge/actions/runs/3552996307
```
Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: azure/setup-helm@v1

The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```
need fixing

Comment (YTGhost) [reply .github/workflows/build-tools.yml:13]: fixed

Comment (gy95) [reply .github/workflows/build-tools.yml:13]: LGTM
reference doc: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/


> @gy95 The ubuntu-18.04 environment is also deprecated, we should switch to ubuntu-20.04 or ubuntu-22.04 instead. see here [actions/runner-images#6002](https://github.com/actions/runner-images/issues/6002)

@fisherxu  PTAL, and another decison, which ubuntu version should we use, ubuntu-20.04 or ubunt-22.04?

**Comment by vincentgoat** [comment]
Thanks!
It's really a great improvement for CI workflows.

**Comment by gy95** [comment]
all warnings fixed, PTAL @fisherxu   and another decison, which ubuntu version should we use, ubuntu-20.04 or ubunt-22.04?

**Comment by fisherxu** [comment]
> all warnings fixed, PTAL @fisherxu and another decison, which ubuntu version should we use, ubuntu-20.04 or ubunt-22.04?

Any difference between ubuntu-20.04 or ubunt-22.04？If there has no special difference, ubunt-22.04 should be okey, wdyt? @gy95

**Comment by gy95** [comment]
22.04: https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md
20.04: https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md

the two linux kernal version are the same, just installing some different version packages, softwares, tools, etc.
I think 22.04 is also OK

**Pull request merged**